src/templates/finops-hub/modules/dataExplorer.bicep

```diff
@@ -0,0 +1,134 @@
+//==============================================================================
+// Parameters
+//==============================================================================
+
+@description('Required. Name of the hub. Used to ensure unique resource names.')
+param hubName string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Resource tags.')
+param tags object = {}
+
+@description('Required. The name of the Azure Key Vault instance.')
+param keyVaultName string
+
+//------------------------------------------------------------------------------
+// Variables
+//------------------------------------------------------------------------------
+
+// Generate globally unique cluster name: 4-22 chars; lowercase letters and numbers
+var safeHubName = replace(replace(toLower(hubName), '-', ''), '_', '')
+var clusterName = replace('${take(safeHubName, 22)}', '--', '-')
+
+//==============================================================================
+// Resources
+//==============================================================================
+
+resource cluster 'Microsoft.Kusto/clusters@2022-12-29' = {
+  name: clusterName
+  location: location
+  tags: tags
+  sku: {
+    capacity: 1
+    name: 'string'
+    tier: 'string'
+  }
+  identity: {
+    type: 'string'
+    userAssignedIdentities: {}
+  }
+  properties: {
+    acceptedAudiences: [
+      {
+        value: 'string'
+      }
+    ]
+    allowedFqdnList: [
+      'string'
+    ]
+    allowedIpRangeList: [
+      'string'
+    ]
+    enableAutoStop: true
+    enableDiskEncryption: false
+    enableDoubleEncryption: false
+    enablePurge: false
+    enableStreamingIngest: false
+    engineType: 'V3'
+    keyVaultProperties: {
+      keyName: keyVaultName
+      keyVaultUri: 'string'
+      keyVersion: 'string'
+      userIdentity: 'string'
+    }
+    languageExtensions: {
+      value: [
+        {
+          languageExtensionImageName: 'string'
+          languageExtensionName: 'string'
+        }
+      ]
+      value: [
+        {
+          languageExtensionImageName: 'string'
+          languageExtensionName: 'string'
+        }
+      ]
+    }
+    optimizedAutoscale: {
+      isEnabled: bool
+      maximum: int
+      minimum: int
+      version: int
+    }
+    publicIPType: 'string'
+    publicNetworkAccess: 'string'
+    restrictOutboundNetworkAccess: 'string'
+    trustedExternalTenants: [
+      {
+        value: 'string'
+      }
+    ]
+    virtualClusterGraduationProperties: 'string'
+    virtualNetworkConfiguration: {
+      dataManagementPublicIpId: 'string'
+      enginePublicIpId: 'string'
+      subnetId: 'string'
+    }
+  }
+  zones: [
+    'string'
+  ]
+}
+
+resource database 'Microsoft.Kusto/clusters/databases@2022-12-29' = {
+  name: 'Hub'
+  location: location
+  parent: cluster
+  kind: 'ReadWrite'
+  properties: {
```

Comment:
How are these used?

Reply:
Michael - I'd set both to "30.00:00:00" (30 days).
Bassel - please correct me if I'm wrong about the format.

Reply:
Bassel, please resolve if you ack. Otherwise, let me know what you think we should set these to and/or if we should make it explicitly configurable by the customer.

Reply:
you can use hotCachePeriodInDays and softDeletePeriodInDays, see https://learn.microsoft.com/en-us/azure/data-explorer/create-cluster-database?tabs=arm